Add tool feature: deployWildFlyApplication

hanjeongsang opened the pull request:
Add tool feature: deployWildFlyApplication #49

jfdenise commented:
@hanjeongsang , that is a very good idea.
I can see some part that you could evolve in your fix:
* The name should be optional. If not provided, use the fileName (eg: app.war). This is what the WildFly CLI is doing. This means that when deploying a directory, a name with the proper extension (.ear, .war, ...) should be provided for the server to identify the nature of the deployment.
* The runtime name should be optional. The server will compute one if not set.
* The archive parameter seems not needed. Just check if the path is a directory or not and you got the archive value.
* The replacement of an existing deployment could be greatly simplified.
    * First check if the deployment exists using its name (I will use the WildFly CLI syntax for the operation): _/deployment=deployment-name:read-resource_.  
    * If the reply has a result, it means that you need to replace it. To do so call: _:full-replace-deployment(name=deployment-name,content=[{path=file-or-directory-path, archive=true_or_false}],enabled=true_ If a runtime-name has been provided, you also need to set it.
    * If the reply has no result, then you need to add the deployment as you did: _/deployment=deployment-name:add(content=[{path=file-or-directory-path,archive=true_or_false}],enabled=true_ If a runtime-name has been provided, you also need to set it.

hanjeongsang commented:
Thank you so much for your detailed feedback.
As you pointed out, for smarter management, I should have minimized the required parameters and let the AI handle the rest automatically, but it seems my approach was not flexible enough.

Honestly, I feel a bit embarrassed by how basic my idea was.
I will make sure to incorporate your feedback into the code.
Have a great day!

hanjeongsang commented:
@jfdenise , I have a question.

Currently, when using the CLI, if I don’t specify a runtime-name for a deployment, it appears that the deployment’s name is used as the runtime-name by default.

I tested this with the following command:
`/deployment=test:add(content=[{path=/APP/deployment/mw-score.war,archive=true}],enabled=true)`

The result of the deployment info command was:
`NAME     RUNTIME-NAME PERSISTENT ENABLED STATUS`
`test     test         true       true    OK`

As you can see, the runtime-name is set to the deployment name. (Tested on WildFly 36.0.1.)

From what I can tell, in JBoss/WildFly,If it doesn't end with *.war, the web context will not be registered by JBoss EAP.

Why doesn’t WildFly automatically use the last part of the path (e.g., the WAR file name) as the runtime-name, instead of just using the deployment name?

The reason I’m curious is that I want to implement a tool that, unlike the CLI’s default behavior, uses the last part of the path as the runtime-name.

hanjeongsang commented:
@jfdenise, thanks for your answer.  but..
What I propose is, when given an exploded deployment directory like /some/path/myapp, how about setting the runtime-name to myapp.war instead of just myapp? Then, the user can access /myapp via browser without any other settings.

hanjeongsang commented:
Okay, good. I'll try modifying the code again as you said. Thank you.

hanjeongsang commented:
@jfdenise, I have reflected the feedback.
I implemented duplicate checking with _**/deployment="deployment name":read-resource**_ and if it already exists, I changed it to replace using  _**:full-replace-deployment()**_.
It works well, and the code has also been simplified.

I have also improved the order of some variables and descriptions in a better direction.
Thank you.

hanjeongsang commented:
@jfdenise, Thank you for your review!
I pressed re-request by mistake. I will correct it and reply again.
